src/views/Search.js

```diff
@@ -1,7 +1,151 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import SearchIcon from '@material-ui/icons/Search';
+import Button from '../components/Button/Button';
+import noImageFound from '../assets/img/no_image_found.png';
+import SingleLineGridList from '../components/Grid/GridList';
+import TextField from '@material-ui/core/TextField';
+import PropTypes from 'prop-types';
+import { search } from '../actions/searchActions';
+import { withStyles } from '@material-ui/core/styles';
 
-const Search = () => {
-  return <h1>Search</h1>;
+const styles = theme => ({
+  container: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    justifyContent: 'flex-end'
+  },
+  textField: {
+    margin: 0,
+    width: 200
+  }
+});
+
+class Search extends Component {
+  state = {
+    searchQuery: ''
+  };
+
+  search = (query, type, limit, offset) => {
+    this.props.search(query, type, limit, offset);
+  };
+
+  /**
+   * size: small, medium, large
+   */
+  getItemImage = (images, minSize, maxSize) => {
+    if (!images || images.length === 0) return noImageFound;
+    const filteredImages = images.filter(
+      ({ width }) => width > minSize && width < maxSize
+    );
+    return filteredImages.length === 0 ? images[0].url : images[0].url;
+  };
+
+  /**
+   * it returns the format required to display data on
+   * SingleLineGridList Component
+   */
+  filterDataToDisplay = items => {
+    if (!items || items.length === 0) return [];
+    return items.map(item => {
+      const { name: title, images, type, album } = item;
+      const imgUrl =
+        type !== 'track'
+          ? this.getItemImage(images, 250, 400)
+          : this.getItemImage(album.images, 250, 400);
+      return {
+        title,
+        imgUrl
+      };
+    });
+  };
+
+  handleChange = event => {
+    this.setState({
+      searchQuery: event.target.value
+    });
+  };
+
+  onSubmit = event => {
+    event.preventDefault();
+    const { searchQuery } = this.state;
+    if (!searchQuery) return;
+    this.search(searchQuery, ['album', 'track', 'artist']);
+  };
+
+  areArraysEmpty = arrays => {
```

Comment:
This should probably moved out to a helper function.

Reply:
Also there's no need to use an arrow function here, as it only depends on it's parameters.

Reply:
Thanks! I'll change that!